modules/drivers/sparksql/src/metabase/driver/hive_like.clj

```diff
@@ -158,6 +160,82 @@
     (recur driver hsql-form (* amount 3) :month)
     (hx/+ (hx/->timestamp hsql-form) (hsql/raw (format "(INTERVAL '%d' %s)" (int amount) (name unit))))))
 
+(defmethod sql.qp/->honeysql [:sparksql :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)]
+    (case unit
+      :year
+      (let [positive-diff
+            (fn [a b]
+              (hx/cast
+               :integer
+               (hx/floor (hx// (hx/- (hsql/call :months_between b a)
+                                     (hx/cast :integer (hsql/call :>
+                                                                  (hsql/call :dayofmonth a)
+                                                                  (hsql/call :dayofmonth b))))
+                               12))))]
+        (hsql/call :case
+                   (hsql/call :<= (hx/->timestamp x) (hx/->timestamp y))
+                   (positive-diff x y)
+                   :else
+                   (hx/* -1 (positive-diff y x))))
+
+      :quarter
+      (let [positive-diff
+            (fn [a b]
+              (hx/cast
+               :integer
+               (hx/floor (hx// (hx/- (hsql/call :months_between b a)
+                                     (hx/cast :integer (hsql/call :>
+                                                                  (hsql/call :dayofmonth a)
+                                                                  (hsql/call :dayofmonth b))))
+                               3))))]
+        (hsql/call :case
+                   (hsql/call :<= (hx/->timestamp x) (hx/->timestamp y))
+                   (positive-diff x y)
+                   :else
+                   (hx/* -1 (positive-diff y x))))
```

Comment:
I would love to see some comments with example SQL output or something for some of the hairier expressions here. Otherwise this is a little tricky to unpack

Reply:
It's a LOT simpler now